aws/rust-runtime/aws-config/src/profile/retry_config.rs

```diff
@@ -106,7 +107,13 @@ impl ProfileFileRetryConfigProvider {
         let selected_profile = match profile.get_profile(selected_profile) {
             Some(profile) => profile,
             None => {
-                tracing::warn!("failed to get selected '{}' profile", selected_profile);
+                // Lambdas don't have home directories and emitting this warning is not helpful
+                // to users running the SDK from within a Lambda. This warning will be silenced
+                // if we determine that that is the case.
+                let is_likely_running_on_a_lambda = check_is_likely_running_on_a_lambda(self);
```

Comment:
I think this is actually a bug for non-Lambda use-cases too. For example, if I'm not using profile credentials on EC2, it's still going to log this warning. To fix this everywhere, I think we should instead check to see if `self.profile_override.is_some()`. That is: if a customer specified a profile name to use, then we should warn if that profile isn't found.